### PR TITLE
update rancid spec and rubric

### LINKED ISSUE #631: Update Rancid Tomatillos Project
# Project to be updated: [Rancid Tomatillos](https://frontend.turing.edu/projects/module-3/rancid-tomatillos-v3.html)

## Summary of changes needed: 
- Currently, the rubric suggests that students will need to demonstrate an understanding of class based components
- The rubric needs to be updated to expect the use of functional components, and the use of the useState and useEffect hooks to manage state and parse updates from the API.
- All group projects should make clear that a failure to participate can result in an individual failing the project, even if the project is otherwise passing
- Update rubrics to use the Yes, Not Yet, Wow scale that was piloted on M1 projects. Use M1/M2 projects as a guide for consistency.

### PR DESCRIPTION
**Pull request by CassandraGoose**
This ticket closes #631 
**Description of changes made**
- remove class related items
- Add functional component and hook related rubric points (though I made the rubric points more open ended than the ticket implied)
- Added the note about participation in the professionalism section
- Updated rancid-tomatillos-v3, since that was the current spec that is on the deployed site. I am happy to have this new version replace the original, if you’d like. 
- Changed wording in learning goals because it was redundant to say that a learning goal is to learn something. 
- Didn’t end up combining deliverables, because the second set of deliverables requires the completion of the first 4 iterations. So, I updated the title of iteration 0 to imply that there’s other deliverables 😂 IDK 😭 
- Updated the Cypress CYOA ideas to incorporate the ideas discussed in the thread here: https://turingschool.slack.com/archives/G39SDD19R/p1684524695077389
- Tried to word rubric points in more open-ended ways so that a student must think critically about how they can achieve competency on their own instead of being told explicitly what to do. I couldn’t think of good ways to do that for all of the bullet points, so only some of them have been changed. I can change these, if you don’t like them! Or, if you have any ideas on wording changes or such, please let me know. 
- Updated the link that used to go to the lifecycle method lesson to now point to https://react.dev/learn/conditional-rendering, since the goal was for them to use conditional rendering and we aren't updating the old lifecycle method lesson
- Updated the router links to go to v6 (https://reactrouter.com/en/main) instead of v5
- Made the api documentation a note instead of a toggle so that it’s searchable. 

**Motivation for any changes**
- seemed like the right thing to do???

**Questions that you have for a reviewer**
- Did y’all want to remove the " **More React and/or Router practice:** - Add a search/filtering functionality" for movies CYOA? I was unsure based on the thread. 

- Let me know what WOW options should be moved to On Track options, because the original rubric had a lot in “Wow” that seemed more like expectations for competency, but I haven't been on Mod 3 in over a year.

- Are there any functional JS items that should be in the rubric that aren’t? I couldn’t remember what functional concepts the students would learn specifically in M3 if any. I didn’t add anything about closures or anything.

### CODE REVIEW

File: projects/module-3/rancid-tomatillos-v3.md [hunk 1]
```diff
@@ -79,11 +79,13 @@ In the future, when we use the actual API to get individual movie details, the i
 
 ### Iteration 3 - Network Requests & Async JS
 
-For the third iteration, we'll refactor our application to use actual data from the database (information is below) instead of our mocked `movieData` file. We'll also add some type checking with [PropTypes](https://frontend.turing.edu/lessons/module-3/proptypes.html)
```
Comment (CassandraGoose) [inline projects/module-3/rancid-tomatillos-v3.md:82]: GitHub is making it look like I deleted the PropTypes note. I didn't it's still in the file. I don't know why this is red.

File: projects/module-3/rancid-tomatillos-v3.md [hunk 2]
```diff
@@ -232,21 +237,19 @@ It is _far_ more important to be very, very solid on the project stated learning
...
 
 Here are some ideas:
 * **More React and/or Router practice:**
   - Add a search/filtering functionality for movies
 * **More testing practice:**
-  - Take time to carefully ensure that you have tested all possible views and user flows
-  - Check to see if you have a test for each user story (using the story's acceptance criteria as a guide)
-  - Supplement your Cypress tests with unit tests and integration tests by using the React Testing Library
+  - Supplement your Cypress tests with a non-trivial amount of unit tests and integration tests by using the React Testing Library 
+  - Implement 2-3 new concepts from the Cypress library that have not been covered in lessons (ie Command)
```
Comment (sertmer) [inline projects/module-3/rancid-tomatillos-v3.md]: Maybe we could link to the cypress docs for these other concepts?

**Review by sertmer — APPROVED** [review]
This was a big update, but I don't have much to add! I think all of these changes help to bring this project in line with the others the students will have seen thus far. I really appreciate the clarity in the `yes` parts of the rubric. They declare our expectations but without just giving them a checklist. Thanks Cass!

File: projects/module-3/rancid-tomatillos-v3.md (hunk 2 above)
Comment (CassandraGoose) [reply projects/module-3/rancid-tomatillos-v3.md]: Love this idea. Will do!

**Pull request merged**